### PR TITLE
Maybe the ShieldArcAbility needs a minor fix?

### PR DESCRIPTION
**Pull request by guiYMOUR**
## Before repair
982a71fa7a68918dc42b711ed8926374
## After repair
image

- when break
image

If your pull request is **not** translation or serverlist-related, read the list of requirements below and check each box:

- [x] I have read the [contribution guidelines](https://github.com/Anuken/Mindustry/blob/master/CONTRIBUTING.md).
- [x] I have ensured that my code compiles, if applicable.
- [ ] I have ensured that any new features in this PR function correctly in-game, if applicable.

### CODE REVIEW

File: core/src/mindustry/entities/abilities/ShieldArcAbility.java [hunk 1]
```diff
@@ -105,6 +116,14 @@ public void init(UnitType type){
 
     @Override
     public void draw(Unit unit){
+        //Display overload progress when reloading the shield
+        Vec2 t = Tmp.v1.set(x, y).rotate(unit.rotation - 90f).add(unit);
+        if(data < 0) {
```
Comment (Anuken) [inline core/src/mindustry/entities/abilities/ShieldArcAbility.java:121]: This shouldn't be here - other types of shields don't have any kind of progress indicator, it's out of place.

File: core/src/mindustry/entities/abilities/ShieldArcAbility.java [hunk 2]
```diff
@@ -79,6 +82,14 @@ public void addStats(Table t){
 
     @Override
     public void update(Unit unit){
+        if(instanceFull) {
+            if (data < 0 && !broken) broken = true;
```
Comment (Anuken) [inline core/src/mindustry/entities/abilities/ShieldArcAbility.java:86]: Incorrect formatting in several places (extra space), see CONTRIBUTING.md.

Comment (guiYMOUR) [reply core/src/mindustry/entities/abilities/ShieldArcAbility.java:86]: OMG, I'm very sorry, I didn't check it carefully. Do I need to delete the excess parts and resend it again?

File: core/src/mindustry/content/Fx.java [hunk 3]
```diff
@@ -2445,6 +2445,29 @@ public class Fx{
...
+        Lines.stroke(3 * e.fout(), e.color);
+        if(e.data instanceof Unit){
+            Unit u = (Unit) e.data;
+            ShieldArcAbility ab = (ShieldArcAbility) Structs.find(u.abilities, a -> a instanceof ShieldArcAbility);
+            if(ab != null){
+                Vec2 pos = Tmp.v1.set(ab.x, ab.y).rotate(u.rotation - 90f).add(u);
+                Lines.arc(pos.x, pos.y, ab.radius + ab.width/2, ab.angle / 360f, u.rotation + ab.angleOffset - ab.angle / 2f);
+                Lines.arc(pos.x, pos.y, ab.radius - ab.width/2, ab.angle / 360f, u.rotation + ab.angleOffset - ab.angle / 2f);
+                for(int i : new int[]{1, -1}){
+                    Seq<Float> lp = new Seq<>();
```
Comment (Anuken) [inline core/src/mindustry/content/Fx.java:2458]: Calling `new` in the main loop anywhere isn't acceptable. Use a temporary static variable if you must..

Comment (guiYMOUR) [reply core/src/mindustry/content/Fx.java]: Thank you for your guidance!
The methods I have been using that I believe are correct may not be suitable. Thank you again.

File: core/src/mindustry/content/Fx.java [hunk 4]
```diff
@@ -2445,6 +2445,27 @@ public class Fx{
         Lines.poly(e.x, e.y, 6, e.rotation + e.fin());
     }).followParent(true),
 
+    arcShieldBreak = new Effect(40, e -> {
+        Lines.stroke(3 * e.fout(), e.color);
+        if(e.data instanceof Unit){
```
[the remark this answers is not in the document]
Comment (guiYMOUR) [reply core/src/mindustry/content/Fx.java:2450]: This is not actually my intention, but the compiler I used when writing this code does not support the use of pattern variables, which caused this result. X-(
image
I will listen to your opinions and try to write code that you approve of using the methods you taught me.

[the remark this answers is not in the document]
Comment (Anuken) [reply core/src/mindustry/content/Fx.java:2450]: > the compiler I used when writing this code does not support the use of pattern variables, which caused this result

You probably don't have the target set to Java 17 with Jabel enabled. See the Java Mod Template buildscript.

[the remark this answers is not in the document]
Comment (guiYMOUR) [reply core/src/mindustry/content/Fx.java]: Thanks!

**Pull request merged**